### ISSUE TITLE
Seafile terminal client doesn't sync on Ubuntu 16.04

### ISSUE THREAD

**Issue by opnoack** [haiwen/seafile#1603]
Greetings!

I'm not sure where to put this issue since there is not repository for the terminal client.

After installing the seafile terminal client on a minimal Ubuntu 16.04 LTS Server no syncing is possible. The progress state in always "unknown" and there is an error in .ccnet/seafile.log:

```
[04/25/16 08:31:24] clone-mgr.c(812): Transition clone state for xxx from [connect] to [connect].
[04/25/16 08:31:24] http-tx-mgr.c(653): libcurl failed to GET https://xxx/seafhttp/protocol-version: Couldn't resolve host name.
```

Software used:
- Ubuntu 16.04 LTS Xenial Xerus
- seafile terminal client 4.3.2 x86-64
- curl 7.47.0-1ubuntu2

Thanks in advance!
Philipp

**Comment by killing** [comment]
It looks like some network issue. Have you setup proxy on the system?

**Comment by opnoack** [comment]
There are now network problems. Everything else is working normally. It is directly connected to the server, no proxy in between.

**Comment by toniwiki** [comment]
Same problem here on Ubuntu 16.04.
Seafile terminal client fail but commands like `curl https://xxx/seafhttp/protocol-version` works. I think it is not a network problem. Any dependencies?

**Comment by pn-d9t** [comment]
Are there any updates, yet? Bug, missing dependencies etc.? I need a solution for this issue.

**Comment by killing** [comment]
We'll check it on 16.04. Sorry we didn't follow up closely with new distros ;-)

**Comment by BytesTuner** [comment]
Same problem here. Seafile terminal client doesn't sync on Ubuntu 16.04 Server.

**Comment by majestixx** [comment]
+1

**Comment by sfPlayer1** [comment]
Building Seafile CLI from source (5.1.2) seems to fix this issue.

**Comment by majestixx** [comment]
That worked for me too.
I used the guide on http://manual.seafile.com/build_seafile/linux.html. However, I additionally had to install openssl-dev

Maybe my binaries are useful to you (without any warranty):
https://pi.majestixx.de/seafile/d/1481b8614c/

**Comment by pn-d9t** [comment]
Could somebody please update the "Seafile Terminal Client" packages on seafile.com?

**Comment by killing** [comment]
It's an incompatibility issue of openssl. We build the package on centos 6. The only solution now is to build from source.
We're working on a more solid solution for packaging seafile client. Please see https://github.com/haiwen/seafile/pull/1686

**Comment by pn-d9t** [comment]
Well, if building from source is the way to go then I will go for it.

@majestixx I followed the compilation guide but how did you build the .deb packages, is there a command line option to do this automatically? I need to distribute the compiled versions on multiple servers.

**Comment by deragon** [comment]
> Package the client for every Linux distribution is no solution.

99% of distributions are derivatives of Fedora and Ubuntu.  Build on those two and you cover a vast majority of distributions, specially the most commonly used.  I think it is worthwhile for you to setup a compile farm for those two distributions and their versions.  With [LXD](https://linuxcontainers.org/lxd) you can easily have multiple versions of Ubuntu available for compilation.

**Comment by deragon** [comment]
BTW, why is only the terminal version affected by this?  Why is the GUI version working?  Can't the terminal version use the same code as the GUI version, stripped of the GUI code?

**Comment by lins05** [comment]
We've setup official PPA (for ubuntu) and official Bintray repo (for Debian) recently. Now you can install the seafile command line client built natively on each linux distro. Check the document [here](https://github.com/haiwen/seafile-user-manual/blob/master/en/desktop/install-on-linux.md).

**Comment by deragon** [comment]
I recently upgraded to 16.04 and to my suprise, seaf-cli 5.0.1 works on it flawlessly.  This issue must occur on more recent version.

The PPA is great, thank you guys.  However, can you confirm that the most recent version fixes this issue? Or should I stick with 5.0.1?
